src/internet_identity/src/activity_stats/event_stats/tests.rs

```diff
@@ -0,0 +1,364 @@
+use crate::activity_stats::event_stats::{
+    update_events_internal, Event, EventData, PrepareDelegationEvent,
+};
+use crate::ii_domain::IIDomain;
+use crate::storage::Storage;
+use crate::DAY_NS;
+use ic_stable_structures::VectorMemory;
+use std::cell::RefCell;
+use std::rc::Rc;
+use std::time::Duration;
+
+const TIMESTAMP: u64 = 1_714_387_451 * 10u64.pow(9);
+const EXAMPLE_URL: &str = "https://example.com";
+const EXAMPLE_URL_2: &str = "https://other-example.com";
+
+#[test]
+fn should_store_event_and_add_to_aggregations() {
```

Comment:
Would it be useful to have a test with different frontend hostnames to check that the buckets are created properly per frontend hostname?